### ISSUE TITLE
Take 5800 hour of 3090 running 100% to run step 1: Step 1: Evaluation Baselines Preparation

### ISSUE THREAD

**Issue by pivtienduc** [mit-han-lab/lmquant#70]
Hi @synxlin 

Is there any way to speed up the compression of Deep Compressor ? With default way of processing it take forever to convert a model.

My PC is Ryzen 5800X, 64 GB Ram and RTX 3090 need nearly 6000 hours to run Evaluation Baselines Preparation.

Thank you so much

**Comment by josephrocca** [comment]
Related:
* https://github.com/mit-han-lab/deepcompressor/issues/66

That said, if your comment is specifically about evaluation - I think it's possible to "skip" it?

IIRC I ran into some issues "actually" skipping it, so ended up just reducing `num_samples` from 5000 to 5:
```
sed -i 's|  num_samples: 5000|  num_samples: 5|g' /workspace/deepcompressor/examples/diffusion/configs/__default__.yaml
```

See [this comment](https://github.com/mit-han-lab/deepcompressor/issues/65#issuecomment-2761726378) by repo owner:

> Yes, you can safely skip step 1 if you don't plan to run the evaluation scripts.

**Comment by pivtienduc** [comment]
> Related:
> 
> * [Batch sizes for 24gb gpu #66](https://github.com/mit-han-lab/deepcompressor/issues/66)
> 
> That said, if your comment is specifically about evaluation - I think it's possible to "skip" it?
> 
> IIRC I ran into some issues "actually" skipping it, so ended up just reducing `num_samples` from 5000 to 5:
> 
> ```
> sed -i 's|  num_samples: 5000|  num_samples: 5|g' /workspace/deepcompressor/examples/diffusion/configs/__default__.yaml
> ```
> 
> See [this comment](https://github.com/mit-han-lab/deepcompressor/issues/65#issuecomment-2761726378) by repo owner:
> 
> > Yes, you can safely skip step 1 if you don't plan to run the evaluation scripts.

Thank you so much for answering my question.  Just one more thing, how do I make deepcompressor change from converting default flux dev model into my finetuned model

**Comment by josephrocca** [comment]
> how do I make deepcompressor change from converting default flux dev model into my finetuned model

One hacky way is to edit the diffusers flux pipeline:
```sh
DIFFUSERS_PATH=$(python -c "import diffusers; print(diffusers.__path__[0])")
nano "$DIFFUSERS_PATH/pipelines/flux/pipeline_flux.py"
```

Specifically you'd want to load the transformer within the `__init__` of `FluxPipeline`:
```py
    def __init__(
        self,
        scheduler: FlowMatchEulerDiscreteScheduler,
        vae: AutoencoderKL,
        text_encoder: CLIPTextModel,
        tokenizer: CLIPTokenizer,
        text_encoder_2: T5EncoderModel,
        tokenizer_2: T5TokenizerFast,
        transformer: FluxTransformer2DModel,
        image_encoder: CLIPVisionModelWithProjection = None,
        feature_extractor: CLIPImageProcessor = None,
    ):
        super().__init__()

        # Add this line (or similar):
        transformer = FluxTransformer2DModel.from_single_file("/workspace/your-custom-flux-dev.safetensors", torch_dtype=torch.bfloat16)
```

**Comment by pivtienduc** [comment]
> > how do I make deepcompressor change from converting default flux dev model into my finetuned model
> 
> One hacky way is to edit the diffusers flux pipeline:
> 
> DIFFUSERS_PATH=$(python -c "import diffusers; print(diffusers.__path__[0])")
> nano "$DIFFUSERS_PATH/pipelines/flux/pipeline_flux.py"
> Specifically you'd want to load the transformer within the `__init__` of `FluxPipeline`:
> 
>     def __init__(
>         self,
>         scheduler: FlowMatchEulerDiscreteScheduler,
>         vae: AutoencoderKL,
>         text_encoder: CLIPTextModel,
>         tokenizer: CLIPTokenizer,
>         text_encoder_2: T5EncoderModel,
>         tokenizer_2: T5TokenizerFast,
>         transformer: FluxTransformer2DModel,
>         image_encoder: CLIPVisionModelWithProjection = None,
>         feature_extractor: CLIPImageProcessor = None,
>     ):
>         super().__init__()
> 
>         # Add this line (or similar):
>         transformer = FluxTransformer2DModel.from_single_file("/workspace/your-custom-flux-dev.safetensors", torch_dtype=torch.bfloat16)

Thank you I will try

**Comment by synxlin** [comment]
Hi @pivtienduc , 

you can specify the path to your fine-tuned model in your command with `--pipeline-path \PATH\TO\YOUR\FLUX\MODEL`. See [here](https://github.com/mit-han-lab/deepcompressor/blob/main/deepcompressor/app/diffusion/pipeline/config.py#L338), if you do not provide `--pipeline-path`, we will use default path based on `--pipeline-name`. These arguments can also be specified in the model yaml config [here](https://github.com/mit-han-lab/deepcompressor/blob/main/examples/diffusion/configs/model/flux.1-dev.yaml#L2).

**Comment by pivtienduc** [comment]
> Hi [@pivtienduc](https://github.com/pivtienduc) ,
> 
> you can specify the path to your fine-tuned model in your command with `--pipeline-path \PATH\TO\YOUR\FLUX\MODEL`. See [here](https://github.com/mit-han-lab/deepcompressor/blob/main/deepcompressor/app/diffusion/pipeline/config.py#L338), if you do not provide `--pipeline-path`, we will use default path based on `--pipeline-name`. These arguments can also be specified in the model yaml config [here](https://github.com/mit-han-lab/deepcompressor/blob/main/examples/diffusion/configs/model/flux.1-dev.yaml#L2).

Hi @synxlin 

If I provide the correct path do I have to change something else to make deepcompressor quantize single model file not diffuser multi checkpoint file (001, 002, 003 things). 

Thank you

**Comment by pivtienduc** [comment]
> > how do I make deepcompressor change from converting default flux dev model into my finetuned model
> 
> One hacky way is to edit the diffusers flux pipeline:
> 
> DIFFUSERS_PATH=$(python -c "import diffusers; print(diffusers.__path__[0])")
> nano "$DIFFUSERS_PATH/pipelines/flux/pipeline_flux.py"
> Specifically you'd want to load the transformer within the `__init__` of `FluxPipeline`:
> 
>     def __init__(
>         self,
>         scheduler: FlowMatchEulerDiscreteScheduler,
>         vae: AutoencoderKL,
>         text_encoder: CLIPTextModel,
>         tokenizer: CLIPTokenizer,
>         text_encoder_2: T5EncoderModel,
>         tokenizer_2: T5TokenizerFast,
>         transformer: FluxTransformer2DModel,
>         image_encoder: CLIPVisionModelWithProjection = None,
>         feature_extractor: CLIPImageProcessor = None,
>     ):
>         super().__init__()
> 
>         # Add this line (or similar):
>         transformer = FluxTransformer2DModel.from_single_file("/workspace/your-custom-flux-dev.safetensors", torch_dtype=torch.bfloat16)

Can you describe how I can implete these codes. Which file I need to copy paste your code. Because I try to read config.py in "D:\AI\Deep\deepcompressor\deepcompressor\app\diffusion\pipeline\config.py" but don't know where to insert those lines and modify those lines.

**Comment by josephrocca** [comment]
@pivtienduc RE-read my comment, talk to an AI about it, spend some more time on it rather than asking for help every 15 minutes on this thread.

**Comment by pivtienduc** [comment]
> [@pivtienduc](https://github.com/pivtienduc) RE-read my comment, talk to an AI about it, spend some more time on it rather than asking for help every 15 minutes on this thread.

I already ask Deepseek AI code for me but somehow deepcompressor run some line then wont run. Ok this time I copy your reply and ask deepseek again

**Comment by pivtienduc** [comment]
> [@pivtienduc](https://github.com/pivtienduc) RE-read my comment, talk to an AI about it, spend some more time on it rather than asking for help every 15 minutes on this thread.

After 2 hour talking with AI, it always make code to use default json file of Flux dev repo and use huggingface model format to quantization, it look to your code and dont have a clue how to use it. It code the config.py in .....\deepcompressor\app\diffusion\pipeline like this:

[config.py.txt](https://github.com/user-attachments/files/19661835/config.py.txt)

No matter how many times I ask AI to fix it. This is the best I can get.

Oh well, it seem like I have to wait for @synxlin to create code to quantize with single file

**Comment by josephrocca** [comment]
If you want to use my hacky fix, then you need to edit this file:
```
DIFFUSERS_PATH=$(python -c "import diffusers; print(diffusers.path[0])")
echo "$DIFFUSERS_PATH/pipelines/flux/pipeline_flux.py"
```
That outputs a path to the specific diffusers file that you need to edit. You need to open that file in your text editor. If you're not using linux, ask an AI to translate those commands to Windows/Mac. This is quite hacky, and depending on your environment you may want to back up the original version first.

Find the `__init__` function like I mentioned, and add the line that I mentioned - the one after "# Add this line (or similar):". You're not pasting the whole code snippet that I gave, you're just adding that single line at the start of the `__init__` function, after the `super().__init__()` line.

You could also just ask an AI for the code to convert your multi-file checkpoint to a single file.

**Comment by pivtienduc** [comment]
> If you want to use my hacky fix, then you need to edit this file:
> 
> ```
> DIFFUSERS_PATH=$(python -c "import diffusers; print(diffusers.path[0])")
> echo "$DIFFUSERS_PATH/pipelines/flux/pipeline_flux.py"
> ```
> 
> That outputs a path to the specific diffusers file that you need to edit. You need to open that file in your text editor. If you're not using linux, ask an AI to translate those commands to Windows/Mac. This is quite hacky, and depending on your environment you may want to back up the original version first.
> 
> Find the `__init__` function like I mentioned, and add the line that I mentioned - the one after "# Add this line (or similar):". You're not pasting the whole code snippet that I gave, you're just adding that single line at the start of the `__init__` function, after the `super().__init__()` line.
> 
> You could also just ask an AI for the code to convert your multi-file checkpoint to a single file.

Thank you so much, I will try again

**Comment by pivtienduc** [comment]
> If you want to use my hacky fix, then you need to edit this file:
> 
> ```
> DIFFUSERS_PATH=$(python -c "import diffusers; print(diffusers.path[0])")
> echo "$DIFFUSERS_PATH/pipelines/flux/pipeline_flux.py"
> ```
> 
> That outputs a path to the specific diffusers file that you need to edit. You need to open that file in your text editor. If you're not using linux, ask an AI to translate those commands to Windows/Mac. This is quite hacky, and depending on your environment you may want to back up the original version first.
> 
> Find the `__init__` function like I mentioned, and add the line that I mentioned - the one after "# Add this line (or similar):". You're not pasting the whole code snippet that I gave, you're just adding that single line at the start of the `__init__` function, after the `super().__init__()` line.
> 
> You could also just ask an AI for the code to convert your multi-file checkpoint to a single file.

Well I already add your line of code. But Deepcompressor still asking for json file. 

When asking AI, it often ask me change a lot of python file which turn deepcompressor into a mess. While I only need telling deepcompressor change from loading multi part of model into just load 1 file only. I believe I'm very close but still don't know how to finish it. 

This is my error log which show Deepcompressor running but still looking for model_index.json which from loading multi part of model not single file.

[Error log 2.txt](https://github.com/user-attachments/files/19725379/Error.log.2.txt)
